packages/server-boilerplate/src/orchestrator/api/ApiBuilder.ts

```diff
@@ -24,13 +29,16 @@ export class ApiBuilder {
 
   private readonly database: TupaiaDatabase;
 
+  private readonly attachSession: (req: Request, res: Response, next: NextFunction) => void;
+
   private attachVerifyLogin?: (req: LoginRequest, res: Response, next: NextFunction) => void;
 
   private verifyLoginMiddleware?: (req: Request, res: Response, next: NextFunction) => void;
 
-  constructor(transactingConnection: TupaiaDatabase) {
+  constructor(transactingConnection: TupaiaDatabase, attachSession = defaultAttachSession) {
```

Comment:
I think I have a slight preference towards setting the `attachSession` on the builder via something like:

```
  private attachSession = defaultAttachSession;
  
  ...
  
  useAttachSession(attachSession: (req: Request, res: Response, next: NextFunction) => void) {
    this.attachSession = attachSession;
  }
```

I appreciate that this could possibly create issues for someone using the `ApiBuilder` who didn't realise you need to specify the custom `attachSession` prior to adding any of their routes, but I feel keeping the constructor cleaner is worth the tradeoff?

Reply:
Ooh yeah I like that actually, and gets in early with the pattern before we have a sprawling list of constructor arguments to support configuration options